book/references/parallel-configuration.md

```diff
@@ -181,3 +183,138 @@ You can get find the directory used on your system by running the following comm
+
+*max_workers_per_node = 1:* Each compute node will only have one worker and only be able to handle one job at a time.
+
+This config will queue 10 Slurm jobs that will run for up to 10 hours each. Each job will use 20 cores and 100GB of RAM on one compute node. Due to the `max_workers_per_node = 1`, each of these Slurm jobs with these resources will be able to handle 1 QIIME 2 action at a time.
+
+### Example Slurm job
+
+This is the Slurm job we submitted that used the above config. We call this job we actually submit directly the "pilot job." This job will itself submit the worker jobs that actually do the work,
+
+```bash
+#!/bin/bash
+
+#SBATCH -e /scratch/<uname>/kraken2/kraken2.err
+#SBATCH -o /scratch/<uname>/kraken2/kraken2.out
+#SBATCH --job-name=kraken2
+#SBATCH --time=24:00:00
+#SBATCH --mem=8G
+
+module load anaconda3
+conda activate q2-shotgun-dev
+
+export TMPDIR=/scratch/<uname>/tmp
+export CACHE=/scratch/<uname>/cache
+
+qiime moshpit classify-kraken2 \
+        --i-seqs "$CACHE:mp-demux" \
+        --i-kraken2-db "$CACHE:workshop-kraken-db" \
+        --p-threads 20 \
+        --p-partitions 10 \
+        --p-memory-mapping false \
+        --o-hits "$CACHE:kraken_hits" \
+        --o-reports "$CACHE:kraken_reports" \
+        --parallel-config /scratch/<uname>/kraken2/conf.toml \
+        --use-cache "$CACHE" \
+        --verbose
+```
+
+`classify-kraken2` is a pipeline that was written specifically to take advantage of QIIME 2's parallel capabilities. It requires a significant amount of compute resources to match a large number of sequences against a very large kraken database hence the large amount of RAM requested in the Parsl config, and the need to run in parallel in the first place.
+
+It will split the input sequences into `--p-num-partitions` (defaults to splitting each sample into its own partition) sets and then classify them in parallel. The 10 partitions here corresponds to our 10 blocks. We will have 10 different sets of sequences each of which can be submitted to its own block. The 20 threads here corresponds to our 20 cores_per_worker. This allows us to classify `num_blocks * workers_per_block * cores_per_worker` or `10 * 1 * 20 = 200` sequences at a time!
+
+We make sure to set our TMPDIR and the [Artifact Cache](artifact-cache-tutorial) we are using for this action to a location that is accessible globally on the HPC we are using. It is important that you do this to make sure your actions which will be spread across compute nodes are wrtiting information that needs to be shared amongst them to a location they can all see.
+
+This job has a walltime of 24 hours which is significantly longer than the jobs we will be submitting. This is because it can take some time after your pilot job starts running for your worker jobs to actually start running. This job also only asks for 8GB of RAM. This is because this job doesn't do any of the actual computation, so it doesn't require a large amount of compute resources.
```

Comment:
```suggestion
This job has a walltime of 24 hours which is significantly longer than the jobs we will be submitting.
This is because it can take some time after your pilot job starts running for your worker jobs to actually start running.
This job also only asks for 8GB of RAM.
This is because this job doesn't do any of the actual computation, so it doesn't require a large amount of compute resources.
```